master/buildbot/scripts/devproxy.py

```diff
@@ -0,0 +1,168 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+
+import asyncio
+import json
+import logging
+
+import aiohttp  # dev-proxy command requires aiohttp! run 'pip install aiohttp'
+import aiohttp.web
+import jinja2
+
+from buildbot.plugins.db import get_plugins
+
+
+class DevProxy:
+    MAX_CONNECTIONS = 10
+
+    def __init__(self, port, next_url, plugins, unsafe_ssl, auth_cookie):
+        while next_url.endswith('/'):
+            next_url = next_url[:-1]
+        self.next_url = next_url
+        self.app = app = aiohttp.web.Application()
+        self.apps = get_plugins('www', None, load_now=True)
+        self.unsafe_ssl = unsafe_ssl
+        cookies = {}
+        if auth_cookie:
+            if "TWISTED_SESSION" in auth_cookie:  # user pasted the whole document.cookie part!
+                cookies = {k: v for k, v in [c.split("=") for c in auth_cookie.split(";")]}
+                auth_cookie = cookies["TWISTED_SESSION"]
+            cookies = {'TWISTED_SESSION': auth_cookie}
+        logging.basicConfig(level=logging.DEBUG)
+        if plugins is None:
+            plugins = {}
+        else:
+            plugins = json.loads(plugins)
+
+        self.plugins = plugins
+
+        app.router.add_route('*', '/ws', self.ws_handler)
+        for path in ['api', 'auth', 'sse', 'avatar']:
+            app.router.add_route(
+                '*', '/%s{path:.*}' % path, self.proxy_handler)
```

Comment:
Readability would be better if this code resided on a single line as it won't exceed the length of existing lines.

Reply:
Also maybe use `.format`?